### PR TITLE
Changed channels and converters to to use QuantityType where units occur

### LINKED ISSUE #331: Units Of Measurement for channels with units
ESH provides the `QuantityType` to convert values to a desired unit (see https://www.eclipse.org/smarthome/documentation/concepts/units-of-measurement.html). The ZigBee binding already makes use of it for the channel-types `measurement_pressure` and `measurement_temperature`. We would like to use QuantityType wherever units occur.

The item-type of the following channel-types would then need to be changed:

channel-type | item-type
------------ | -------------
battery_voltage | Number:ElectricPotential
electrical_activepower | Number:Power
electrical_rmscurrent | Number:ElectricCurrent
electrical_rmsvoltage | Number:ElectricPotential
measurement_illuminance | Number:Illuminance

### PR DESCRIPTION
**Pull request by TomTravaglino**
Resolves #331 

The item-type of following channels have been changed:

channel-type               | item-type
------------------------------------------------------
battery_voltage 	   | Number:ElectricPotential
electrical_activepower 	   | Number:Power
electrical_rmscurrent 	   | Number:ElectricCurrent
electrical_rmsvoltage 	   | Number:ElectricPotential
measurement_illuminance    | Number:Illuminance

Signed-off-by: Tommaso Travaglino <tommaso.travaglino@telekom.de>

### CODE REVIEW

File: org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/internal/converter/ZigBeeConverterIlluminance.java [hunk 1]
```diff
@@ -86,7 +89,8 @@ public void attributeUpdated(ZclAttribute attribute) {
                 && attribute.getId() == ZclIlluminanceMeasurementCluster.ATTR_MEASUREDVALUE) {
             Integer value = (Integer) attribute.getLastValue();
             if (value != null) {
-                updateChannelState(new DecimalType(BigDecimal.valueOf(value, 2)));
+                updateChannelState(new QuantityType<Illuminance>(BigDecimal.valueOf(value, 2),
```
Comment (hsudbrock) [inline org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/internal/converter/ZigBeeConverterIlluminance.java]: I had a look in to the cluster library spec (4.2.2.1.1), and it states that the measured value is *not* the illuminance in Lux, but rather `measuredValue = 10.000 x log10(Illuminance) + 1`. Hence, quantifying the measured value in Lux is probably not correct here.

File: org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/internal/converter/ZigBeeConverterMeasurementRmsVoltage.java [hunk 2]
```diff
@@ -134,7 +138,8 @@ public void attributeUpdated(ZclAttribute attribute) {
                 && attribute.getId() == ZclElectricalMeasurementCluster.ATTR_RMSVOLTAGE) {
             Integer value = (Integer) attribute.getLastValue();
             if (value != null) {
-                updateChannelState(new DecimalType(value * multiplier / divisor));
+                updateChannelState(new QuantityType<ElectricPotential>(BigDecimal.valueOf(value * multiplier / divisor),
```
[the remark this answers is not in the document]
Comment (TomTravaglino) [reply org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/internal/converter/ZigBeeConverterMeasurementRmsVoltage.java]: Will change it

File: org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/internal/converter/ZigBeeConverterIlluminance.java (hunk 1 above)
Comment (TomTravaglino) [reply org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/internal/converter/ZigBeeConverterIlluminance.java]: Looked at the spec too and agree that apparently the measured value is not in Lux. Will revoke this change and restore the previous version.

**Comment by cdjackson** [comment]
Please do not merge this as I prefer not to add this before 2.4 is released.

**Comment by cdjackson** [comment]
Thanks Henning. 2.4 should be released next week and we should now not be adding new features. Once it’s done then I’ll take a look.

**Comment by hsudbrock** [comment]
@TomTravaglino Sorry, one thing I did not see before: You use `%unit` in the channel type configuration, but this is not used in the channel types that already use units of measurement (like temperature). Should those be changed as well with this PR (so that all channel types are in sync with respect to this), or should your changes be aligned with how this was handled before this PR, or is it intended that temperature is handled differently?

**Comment by TomTravaglino** [comment]
@hsudbrock According to the ESH documentation the %unit% placeholder is needed to display the unit. Will change `measurement_temperature` and `measurement_pressure` to contain the placeholder too.

**Comment by hsudbrock** [comment]
@cdjackson From my side, you could merge this after the release has been done (but, of course, feel free to also review if you like).

**Comment by cdjackson** [comment]
LGTM - I will merge once 2.4 is released.

**Comment by cdjackson** [comment]
Thanks Tammaso

**Pull request merged**